bin/src/data/csv.py

```diff
@@ -223,29 +222,48 @@ def get_and_encode(self, dictionary: dict, idx: Any) -> dict:
             # get the data at the given index
             # if the data is not a list, it is converted to a list
             # otherwise it breaks Float().encode_all(data) because it expects a list
-            data = dictionary[key][idx]
+            if idx is None:
+                data = dictionary[key]
+            else:
+                data = dictionary[key][idx]
             if not isinstance(data, list):
                 data = [data]
 
             # check if 'data_type' is in the experiment class attributes
             if not hasattr(self.experiment, data_type.lower()):
                 raise ValueError(f"The data type {data_type} is not in the experiment class attributes. the column name is {key}, the available attributes are {self.experiment.__dict__}")
-            
+
             # encode the data at given index
             # For that, it first retrieves the data object and then calls the encode_all method to encode the data
-            output[name] = self.experiment.get_function_encode_all(data_type)(dictionary[key][idx])
+            output[name] = self.experiment.get_function_encode_all(data_type)(data)
 
         return output
-    
+
+    def get_all_items(self) -> Tuple[dict, dict, dict]:
+        """
+        Returns all the items in the csv file, encoded.
+        TODO in the future we can optimize this for big datasets (ie. using batches, etc).
+        """
+        return self.get_and_encode(self.input), self.get_and_encode(self.label), self.meta
+
+    def get_all_items_and_length(self) -> Tuple[dict, dict, dict, int]:
```

Comment:
why can not be there a different function to cumppute length? is there a reason id done here?